core/primitives-core/src/runtime/fees.rs

```diff
@@ -203,22 +204,29 @@ impl StorageUsageConfig {
 }
 
 /// Helper functions for computing Transfer fees.
-/// In case of implicit account creation they always include extra fees for the CreateAccount and
+/// In case of implicit account creation they include extra fees for the CreateAccount and
 /// AddFullAccessKey (for NEAR-implicit account only) actions that are implicit.
 /// We can assume that no overflow will happen here.
 pub fn transfer_exec_fee(
     cfg: &RuntimeFeesConfig,
     implicit_account_creation_allowed: bool,
     receiver_account_type: AccountType,
+    protocol_version: ProtocolVersion,
 ) -> Gas {
     let transfer_fee = cfg.fee(ActionCosts::transfer).exec_fee();
     match (implicit_account_creation_allowed, receiver_account_type) {
         // Regular transfer to a named account.
         (_, AccountType::NamedAccount) => transfer_fee,
         // No account will be created, just a regular transfer.
         (false, _) => transfer_fee,
-        // Currently, no account is created on transfer to ETH-implicit account, just a regular transfer.
-        (true, AccountType::EthImplicitAccount) => transfer_fee,
+        // Extra fee for the CreateAccount.
+        (true, AccountType::EthImplicitAccount) => {
+            if checked_feature!("stable", EthImplicit, protocol_version) {
+                transfer_fee + cfg.fee(ActionCosts::create_account).exec_fee()
```

Comment:
Why is it acceptable to use unchecked arithmetic?
`Gas` is simply a `u64`.

Reply:
@nikurt
It is a pretty small constant that we know is within bounds.
But nothing stops us for using checked arithmetic here.
Although it seems we use unchecked arithmetic for calculating fees in many places, I tried to follow the existing code. Wdyt?

Reply:
Your code is consistent with the surrounding code and I don't expect you to rework this whole module.
The argument that gas costs behave nicely is a good enough argument.

Reply:
Found this documentation on the topic:
https://github.com/near/nearcore/blob/master/docs/practices/style.md#arithmetic-integer-operations